solutions/communication/index.asciidoc

```diff
@@ -0,0 +1,108 @@
+
+Here we try to expose cross-cutting best practices for communication among services in larger IT application landscapes on a high level. Details how to implement this with specific libraries or programming-languages are described in the individual stacks of devonfw.
+
+== Communication types
+Generally, types of communication can be classified in 2 axes.
+
+The first axis defines if the protocol is synchronous or asynchronous:
+
+    * *Synchronous protocol*: The client sends a request and waits for a response from the service as the client code can only continue its task when it receives the server response. That's independent of the client code execution that could be synchronous (thread is blocked) or asynchronous (thread isn't blocked, and the response will reach a callback eventually). e.g. HTTP/HTTPs
+    * *Asynchronous protocol*: The client code or message sender usually doesn't wait for a response. It just sends the message as when sending a message to a queue or any other message broker. e.g AMQP
+
+The second axis defines if the communication has a single receiver or multiple receivers:
+
+    * *Single receiver*: Each request must be processed by exactly one receiver or service.
+    * *Multiple receivers*: Each request can be processed by zero to multiple receivers. This type of communication must be asynchronous. An example is the publish/subscribe mechanism.
+
+One common style is single-receiver communication with a synchronous protocol like HTTP/HTTPS when invoking a regular Web API HTTP service. Microservices also typically use messaging protocols for asynchronous communication between microservices. We will discuss more about some popular approaches adopted these days including:
+
+    * <<rpc>>
+    * <<messaging_and_eventing>>
+    * <<service_mesh>>
+
+[#rpc]
+== RPC
+When it comes to RPC communication nowadays we immediately talk about HTTPS.
+
+The most common choice is https://en.wikipedia.org/wiki/Representational_state_transfer[REST] with https://www.json.org/[JSON] that perfectly works together with web browsers.
+However, please also consider other options especially for backend communication:
+
+* https://grpc.io/[gRPC] is an efficient, high performance RPC protocol
+* ...
+
+Protocols like https://en.wikipedia.org/wiki/SOAP[SOAP] or even RMI, Corba, etc. should be considered as legacy and discouraged.
+
+[#messaging_and_eventing]
+== Messaging and eventing
+
+We do not even try to distinguish between _messaging_ and _eventing_.
+A message can be seen as an event and an event as a message.
+The nature of this communication style is that it is entirely asynchronous and the sender of the event or message does not have to know about the receipient(s).
```

Comment:
Should we add an image here?

Reply:
I prefer not. It is quite a general introduction